2_process.R

```diff
@@ -61,5 +62,40 @@ p2_targets_list <- list(
         purrr::map(.,summarize_paired_comids) %>%
         bind_rows()
     }
-  )
+  ),
+  
+  # Extract COMID and respective area for LC estimation below
+  tar_target(p2_nhd_area_att, 
+             p1_nhdv2reaches_sf %>%
+               st_drop_geometry() %>%
+               select(COMID,AREASQKM,TOTDASQKM)
+  ),
+  
+  ## Melt PRMS_nhdv2_xwalk to get all cols of comids Ids filtered to drb 
+  tar_target(p2_drb_comids, 
```

Comment:
Consider simplifying this target and possible combining with the `read_subset()` function when creating `p1_NLCD_df` target

Reply:
> Hmm, this target reminds me of your question of when should we put raw code in the target object itself versus a separate function. I know I wrote a version of this code but now as a reader it seems hard to quickly interpret the 2_process file 😸. Maybe we could simplify this target by replacing the split-lapply code with the line below. Does that work (i.e. give the same output)?

> p2_prms_nhdv2_xwalk %>% tidyr::separate_rows(comid_all,sep=";")

Nice job finding this function! Exactly what I am looking for, had not yet found this simple one-liner for this tidy process.  🥳

Reply:
> I could see the value of using this target as input to read_subset_LC_data. Currently that function takes all the COMIDs within the whole area of interest (~approximately the bbox of the DRB), which leaves us with more COMIDs (and thus more LC values) than we really need. However, the xwalk target that p2_drb_comids stems from gets generated within the 2_process phase, whereas the LC data is initially subset in the 1_fetch phase. Were you thinking of moving one or both targets to combine with the read_subset_LC_data function?

Yea, these are exactly the questions I was thinking about. It requires some decision on what we include in our pipeline's fetch and process. Computationally, its not adding too much to conduct two subsets. Also, I initially saw these two subsets as different: the *1st* in fetch subsets the raw LC data that originally covers all of Conus, so its easier to work with; the *2nd* is to join comid area attributes and to ensure the scaling to PRMS is possible.